gsy_e_sdk/setups/asset_api_scripts/redis_basic_strategies.py

```diff
@@ -0,0 +1,196 @@
+                                     2 * self.asset_strategy[area_uuid]["fee_to_market_maker"] -
+                                     fit_rate) * (tick / TICKS)
+                                    )
+                        load_strategy.append(buy_rate)
+                    else:
+                        buy_rate = (market_maker_rate +
+                                    self.asset_strategy[area_uuid]["fee_to_market_maker"])
+                        load_strategy.append(buy_rate)
+                self.asset_strategy[area_uuid]["buy_rates"] = load_strategy
+
+            # Generation strategy
+            if "available_energy_kWh" in area_dict["asset_info"]:
+                gen_strategy = []
+                for tick in range(0, TICKS):
+                    if tick < TICKS - 2:
+                        sell_rate = (market_maker_rate +
+                                     self.asset_strategy[area_uuid]["fee_to_market_maker"] -
+                                     (market_maker_rate +
+                                      2 * self.asset_strategy[area_uuid]["fee_to_market_maker"] -
+                                      fit_rate) * (tick / TICKS)
+                                     )
+                        gen_strategy.append(max(0, sell_rate))
+                    else:
+                        sell_rate = fit_rate - (
+                            self.asset_strategy[area_uuid]["fee_to_market_maker"])
+                        gen_strategy.append(max(0, sell_rate))
+                self.asset_strategy[area_uuid]["sell_rates"] = gen_strategy
+
+            # Storage strategy
+            if "used_storage" in area_dict["asset_info"]:
+                batt_buy_strategy = []
+                batt_sell_strategy = []
+                for tick in range(0, TICKS):
+                    buy_rate = (fit_rate -
+                                self.asset_strategy[area_uuid]["fee_to_market_maker"] +
+                                (med_price -
+                                 (fit_rate -
+                                  self.asset_strategy[area_uuid]["fee_to_market_maker"]
+                                  )
+                                 ) * (tick / TICKS)
+                                )
+                    batt_buy_strategy.append(buy_rate)
+                    sell_rate = (market_maker_rate +
+                                 self.asset_strategy[area_uuid]["fee_to_market_maker"] -
+                                 (market_maker_rate +
+                                  self.asset_strategy[area_uuid]["fee_to_market_maker"] -
+                                  med_price) * (tick / TICKS)
+                                 )
+                    batt_sell_strategy.append(sell_rate)
+                self.asset_strategy[area_uuid]["buy_rates"] = batt_buy_strategy
+                self.asset_strategy[area_uuid]["sell_rates"] = batt_sell_strategy
+
+    def post_bid_offer(self, rate_index=0):
```

Comment:
Sorry, I just see this now. The `rate_index` is always 0. So only the first value of the `self.asset_strategy` is used. Is this wanted? Why are you constructing the `self.asset_strategy` with 10 members in the first place?
Also, your explanation for the `TICKS` constant does not conform to this behaviour. Or did I miss something?

Reply:
Answer from Andrea:

> Actually, rate_index = 0 only when I call the post_bid_offer function without specifying any argument, as in the on_market_cycle method. However, every time on_tick is triggered, I call the post_bid_offer function specifying the rate_index, overwriting the default value of 0. Thus, all the 10 members of the strategy are actually used.

Reply:
I see now. Thanks for pointing me to it. 
Now I understand fully what you are trying here. Unfortunately, this implementation only works, if TICK=10. 
If we change the frequency of the dispatching of the `tick_info` in the backend from every 10% to e.g. every 5 %, this script will fail, because e.g. 80/5 = 16 (which wich is an index that will not part of your `asset_strategy` dictionary then. 
Also, if the user decides to post offers more often then every 10% of the slot, he/she can not simple change the value. This is probably why you added `to leave as it is` to the comment, right? 
So, as this is a backend-dependant setting, I would vote for renaming it to: `TICK_DISPATCH_FREQUENCY_PERCENT` or similar.
Sorry to be relentless here, but the point of these refactorings was also to be better debuggable by the backend team.